sorc/chgres_cube.fd/program_setup.f90

```diff
@@ -293,6 +369,41 @@ subroutine read_setup_namelist
    case default
      call error_handler("UNRECOGNIZED INPUT DATA TYPE.", 1)
  end select
+
+!-------------------------------------------------------------------------
+! Ensure proper file variable provided for grib2 input  
+!-------------------------------------------------------------------------
+
+ if (trim(input_type) == "grib2") then
+	 if (trim(grib2_file_input_grid) == "NULL" .or. trim(grib2_file_input_grid) == "") then
+		 call error_handler("FOR GRIB2 DATA, PLEASE PROVIDE GRIB2_FILE_INPUT_GRID", 1)
+	 endif
+ endif
+
+ !-------------------------------------------------------------------------
+! For grib2 input, warn about possibly unsupported external model types
+!-------------------------------------------------------------------------
+
```

Comment:
I would stop processing here.  Other grib2 data will almost certainly not work.

Reply:
@GeorgeGayno-NOAA, @LarissaReames-NOAA and I wanted to provide the option for users who are advanced enough to try their own GRIB2 files, given that GRIB2 files are very standardized and have a better chance of working than other kinds of initialization datasets.  In addition, we provide the option for users to create their own varmap tables, so it would make sense that they could in turn try new data sources.  We do warn the user of the supported GRIB2 files to this point, so they should be aware that what they're trying may not work immediately.

Reply:
GRIB2 is a standard data format, but the contents of a GRIB2 file is certainly not standard.  The surface component of chgres_cube is very Noah-LSM-centric.  Ingesting land states from models other than Noah will not work well.  A novice user may get chgres to run without error with his GRIB2 data, but the initial land states may not be what he expects.  Land states require a lot of special handling.  It is more involved than, say, using 500 mb temperature from other dynamic core.

Reply:
Could you point us to the places in the code that are Noah-LSM centric? I'm not quite sure that part is clear to me. I might be able to provide a better response once I understand that distinction.

Reply:
@GeorgeGayno-NOAA My understanding from discussions with @LarissaReames-NOAA is that the soil and vegetation classifications for RUC-LSM and NOAH LSM are identical.  We've tested both RAP and HRRR data (that use RUC-LSM) extensively with this new code and are confident that we're providing FV3 with the correct soil levels/values, etc.  However, as you mention, it is possible that a user may try a new, unknown LSM that would be incompatible or may not provide results that are expected.  What if we cause a failure for all unknown GRIB2 source data, but then tell users that they may try to use the data **at their own risk** by modifying the code?

Reply:
@GeorgeGayno-NOAA The latest commit pushed includes a change to fail if the user chooses an unsupported grib2 file, which instructions to proceed at their own risk, if they wish.

Reply:
> Could you point us to the places in the code that are Noah-LSM centric? I'm not quite sure that part is clear to me. I might be able to provide a better response once I understand that distinction.

Here are the EMC Noah-LSM-centric parts of the code:

Routine 'calc_soil_params_driver' computes soil thresholds, such as wilting point, according to the soil type and the soil hydraulic coefficients.  The coefficients are those use by EMC's Noah-lsm.  I can't say what thresholds other LSMs use.

The above soil thresholds are used in 'rescale_soil_moisture' to rescale the soil moisture based on differences between the input and target grids.  The method is specifically implemented for the Noah-lsm.  Is it applicable to other LSMs?  I don't know.

The function FHR20 computes the frozen portion of the total soil moisture.  This function is part of the Noah-lsm. I don't know what other land models use to compute frozen soil.

Reply:
@GeorgeGayno-NOAA @LarissaReames-NOAA, I have not heard back from Tanya regarding whether the equations used in these routines are equivalent to what RUC-LSM does or whether we will need to incorporate different calculations for RUC-LSM vs NOAH based on the number of soil levels requested (9 for RUC-LSM, 4 for NOAH).  Given the code slush on Wednesday, Sept. 30, can we merge this PR and create a UFS_UTILS release/public-v2 branch now?  Then we can open a PR to both develop and release/public-v2 if we have these RUC-LSM changes to merge later.  We're up against this deadline to make sure the code is ready to start porting to other platforms on which we've not yet tested, and it would be good to get that started ASAP.  Thanks.